### PR TITLE
[WFLY-12901] Add a new quickstart covering MicroProfile JWT.

### PR DESCRIPTION
**Pull request by darranl**
Includes a complete example and instructions to create the example from scratch.
Also contains a test case to enable testing the interaction with the example.

https://issues.redhat.com/browse/WFLY-12901

Analysis - https://github.com/wildfly/wildfly-proposals/pull/273

Depends on https://github.com/wildfly/wildfly-archetypes/pull/6

### CODE REVIEW

**Comment by sschmiedleitner** [comment]
After reviewing this PR I have some concerns...

I would have expected to get an idea, how to use Wildfly 19 with a deployed microprofile application, that just is secured by JWT. The JWT is not issued by this wildfly but rather by an external keycloak server (at least I have such a setup).

But your PR indicates, that there is a very special configuration of Elytron needed (issuer, keystore) which I don't have with an external authentication server like keycloak. It should rather use the settings of the microprofile.properties file:

mp.jwt.verify.issuer=<xy>
mp.jwt.verify.publickey[.location]=<xyz>

How can I manage this?

**Comment by darranl** [comment]
The reason this as a quickstart is not using an external issuer is that it would be a lot of additional complexity just to get the quickstart running when the token issuing can be simplified.  

Overall the aim of this quickstart is to demonstrate how to enable MicroProfile JWT authentication for a deployment deployed to WildFly it was not about setting up the complete environment, projects such as KeyCloak would then provide their own getting started material to start to use them as an issuer.

I am not really sure what you mean with this line: -

> But your PR indicates, that there is a very special configuration of Elytron needed (issuer, keystore)

This pull request does not mention Elytron once, this quickstart exclusively makes use of microprofile config properties for configuration.

**Comment by sschmiedleitner** [comment]
Hm, Elytron is mentioned in the chapter "Configure the Server" serveral times and also refers to configure-elytron.cli.

I can follow the approach that a quickstart should not have too much external dependencies - I aggree. 

Still I would like to get my setup running somehow and I am serching for input how to configure Wildfly 19 properly to use my configuration from microprofile.properties and @LoginConfig

**Comment by darranl** [comment]
This may be a bit clearer now, the main files were missing from this pull request.

**Comment by sschmiedleitner** [comment]
I am really sorry, I was looking at the "jaxrs-jwt" quickstart, not your new one...

With the correct "microprofile-jwt" I was able to figure out that I need to set the "resteasy.role.based.security" configuration properly!

**Comment by rhusar** [comment]
> Depends on wildfly/wildfly-archetypes#6

Might wanna update that since it was closed by @maeste.

**Comment by darranl** [comment]
@rhusar Yes that is the plan to rewrite using a generic archetype.

**Comment by darranl** [comment]
@emmartins This quickstart is now also updated so all of it's dependencies beyond "wildfly-jakartaee8-with-tools" are now included in it's pom without modifications to the parent pom.

File: microprofile-jwt/README.adoc [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+
+
+@Path("/Sample")
+public class SampleEndPoint {
+
+    @GET
+	@Path("/helloworld")
+	public String helloworld() {
+        return "Hello World";
+	}
```
Comment (rhusar) [inline microprofile-jwt/README.adoc]: Indentation is off here.

File: microprofile-jwt/pom.xml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>20.0.0.Beta1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>microprofile-jwt</artifactId>
+    <packaging>war</packaging>
+    <name>JBoss PRODUCT_TYPE Quickstart: MicroProfile JWT</name>
```
Comment (rhusar) [inline microprofile-jwt/pom.xml]: The standard module names are "Quickstart: artifactId", e.g. https://github.com/wildfly/quickstart/blob/master/contacts-jquerymobile/pom.xml#L33

Comment (darranl) [reply microprofile-jwt/pom.xml]: This format comes directly from the template https://github.com/wildfly/quickstart/blob/master/template/pom.xml#L25

File: microprofile-jwt/README.adoc (hunk 1 above)
Comment (darranl) [reply microprofile-jwt/README.adoc]: Fixed

File: microprofile-jwt/pom.xml (hunk 2 above)
Comment (rhusar) [reply microprofile-jwt/pom.xml]: Yeah, looks like the template is outdated, i.e. 

:-)

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Quickstart: Parent 20.0.0.Beta1-SNAPSHOT:
[INFO]
[INFO] Quickstart: Parent ................................. SUCCESS [  0.090 s]
[INFO] Quickstart: app-client ............................. SUCCESS [  0.002 s]
[INFO] Quickstart: app-client - ejb ....................... SUCCESS [  0.018 s]
[INFO] Quickstart: app-client - client-simple ............. SUCCESS [  0.010 s]
[INFO] Quickstart: app-client - ear ....................... SUCCESS [  0.029 s]
[INFO] Quickstart: batch-processing ....................... SUCCESS [  0.036 s]
[INFO] Quickstart: bean-validation ........................ SUCCESS [  0.017 s]
...
[INFO] JBoss PRODUCT_TYPE Quickstart: MicroProfile JWT .... SUCCESS [  0.002 s]
[INFO] Quickstart: numberguess ............................ SUCCESS [  0.010 s]
[INFO] Quickstart: payment-cdi-event ...................... SUCCESS [  0.015 s]
[INFO] Quickstart: resteasy-jaxrs-client .................. SUCCESS [  0.011 s]
[INFO] Quickstart: security-domain-to-domain .............. SUCCESS [  0.003 s]
[INFO] Quickstart: security-domain-to-domain - ejb ........ SUCCESS [  0.010 s]
[INFO] Quickstart: security-domain-to-domain - web ........ SUCCESS [  0.015 s]
[INFO] Quickstart: security-domain-to-domain - ear ........ SUCCESS [  0.005 s]
[INFO] Quickstart: servlet-async .......................... SUCCESS [  0.015 s]
[INFO] Quickstart: servlet-filterlistener ................. SUCCESS [  0.013 s]
[INFO] Quickstart: servlet-security ....................... SUCCESS [  0.013 s]
[INFO] Quickstart: shopping-cart .......................... SUCCESS [  0.001 s]
[INFO] Quickstart: shopping-cart - server ................. SUCCESS [  0.008 s]
[INFO] Quickstart: shopping-cart - client ................. SUCCESS [  0.008 s]
[INFO] Quickstart: spring-greeter ......................... SUCCESS [  0.019 s]
[INFO] Quickstart: spring-kitchensink-basic ............... SUCCESS [  0.025 s]
[INFO] Quickstart: spring-kitchensink-springmvctest ....... SUCCESS [  0.022 s]
[INFO] Quickstart: spring-resteasy ........................ SUCCESS [  0.012 s]
[INFO] Quickstart: tasks-jsf .............................. SUCCESS [  0.016 s]
[INFO] Quickstart: tasks-rs ............................... SUCCESS [  0.015 s]
[INFO] Quickstart: temperature-converter .................. SUCCESS [  0.012 s]
[INFO] Quickstart: thread-racing .......................... SUCCESS [  0.028 s]
[INFO] Quickstart: websocket-client ....................... SUCCESS [  0.013 s]
[INFO] Quickstart: websocket-endpoint ..................... SUCCESS [  0.014 s]
[INFO] Quickstart: websocket-hello ........................ SUCCESS [  0.009 s]
[INFO] Quickstart: wicket-ear ............................. SUCCESS [  0.001 s]
[INFO] Quickstart: wicket-ear - ejb ....................... SUCCESS [  0.009 s]
[INFO] Quickstart: wicket-ear - war ....................... SUCCESS [  0.019 s]
[INFO] Quickstart: wicket-ear - ear ....................... SUCCESS [  0.004 s]
[INFO] Quickstart: wicket-war ............................. SUCCESS [  0.019 s]
[INFO] Quickstart: xml-jaxp ............................... SUCCESS [  0.018 s]
[INFO] Quickstart: jts .................................... SUCCESS [  0.002 s]
[INFO] Quickstart: jts - application-component-2 .......... SUCCESS [  0.012 s]
[INFO] Quickstart: jts - application-component-1 .......... SUCCESS [  0.014 s]
[INFO] Quickstart: ejb-remote ............................. SUCCESS [  0.001 s]
[INFO] Quickstart: ejb-remote - server-side ............... SUCCESS [  0.007 s]
[INFO] Quickstart: ejb-remote - client .................... SUCCESS [  0.010 s]
[INFO] Quickstart: jta-crash-rec .......................... SUCCESS [  0.011 s]
[INFO] Quickstart: wsat-simple ............................ SUCCESS [  0.016 s]
[INFO] Quickstart: wsba-coordinator-completion-simple ..... SUCCESS [  0.012 s]
[INFO] Quickstart: wsba-participant-completion-simple ..... SUCCESS [  0.012 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.461 s
[INFO] Finished at: 2020-03-17T14:37:54+01:00
[INFO] ------------------------------------------------------------------------
```

Comment (rhusar) [reply microprofile-jwt/pom.xml]: Opened https://github.com/wildfly/quickstart/pull/392 with the tempate fix.

**Pull request merged**